ktor-features/ktor-metrics-micrometer/jvm/src/io/ktor/metrics/micrometer/MicrometerMetrics.kt

```diff
@@ -0,0 +1,221 @@
+ * <ul>
+ *     <li><code>ktor.http.server.requests.active</code>: Gauge - The amount of active ktor requests</li>
+ *     <li><code>ktor.http.server.requests</code>: Timer - Timer for all requests. By default no percentiles or
+ *       histogram is exposed. Use the [Configuration.distributionStatisticConfig] to enable these.
+ *       Tags by default (use [Configuration.tags] to configure the tags or add custom tags):
+ *       <ul>
+ *           <li><code>local</code>: The host and port of the request uri (e.g. 'www.ktor.io')</li>
+ *           <li><code>method</code>: The http method (e.g. 'GET')</li>
+ *           <li><code>route</code>: The use ktor route used for this request. (e.g. '/some/path/{someParameter}')
+ *           <li><code>status</code>: The http status code or the Exception name thrown during the request (e.g. '200', 'java.lang.IllegalArgumentException')
+ *        <ul>
+ *     <li>
+ *  <ul>
+ */
+class MicrometerMetrics(
+    private val registry: MeterRegistry,
+    timerDistributionConfig: DistributionStatisticConfig,
+    private val tags: MicrometerMetrics.TagBuilder.() -> Unit
+) {
+
+    private val active = registry.gauge(activeGaugeName, AtomicInteger(0))
+
+    init {
+        enableTimerDistributionConfig(timerDistributionConfig)
+    }
+
+    private fun enableTimerDistributionConfig(timerDistributionConfig: DistributionStatisticConfig) {
+        registry.config().meterFilter(object : MeterFilter {
+            override fun configure(id: Meter.Id, config: DistributionStatisticConfig): DistributionStatisticConfig =
+                if (id.name == requestTimerName)
+                    timerDistributionConfig.merge(config)
+                else
+                    config
+        })
+    }
+
+    /**
+     * Configures this Feature
+     * @property registry The meter registry where the meters are registered. Mandatory
+     * @property meterBinders The binders that are automatically bound to the registry. Default: [ClassLoaderMetrics],
+     * [JvmMemoryMetrics], [ProcessorMetrics], [JvmGcMetrics], [ProcessorMetrics], [JvmThreadMetrics], [FileDescriptorMetrics]
+     * @property distributionStatisticConfig configures the histogram and/or percentiles for all request timers. By
+     * default 50%, 90% , 95% and 99% percentiles are configured
+     * @property tags is called to generate the tags for this call. By default, [TagBuilder.defaultTags] are generated
+     */
+    class Configuration {
+
+        lateinit var registry: MeterRegistry
+
+        var meterBinders: List<MeterBinder> = listOf(
+            ClassLoaderMetrics(),
+            JvmMemoryMetrics(),
+            JvmGcMetrics(),
+            ProcessorMetrics(),
+            JvmThreadMetrics(),
+            FileDescriptorMetrics()
+        )
+
+        var distributionStatisticConfig: DistributionStatisticConfig =
+            DistributionStatisticConfig.Builder()
+                .percentiles(0.5, 0.9, 0.95, 0.99)
```

Comment:
I would discourage client-side percentiles, especially for request metrics, since they are non-aggregable. They really are only usable in the narrow case where a timer/summary has a really small total tag cardinality such that you can reasonably plot all the tag variation in one chart (e.g. a timer with no tags other than `instanceId` running in a cluster with <10 instances).

Reply:
👍 for that argument, I agree that it does not make that much sense in productive environments, but not all micrometer backends support histograms, I'd rather provide some client side percentiles as a starter than nothing at all. But we should document that we encourage to provide histograms.

Reply:
It's easy to disable them, right?